index.ts

```diff
@@ -252,7 +264,7 @@ wss.on("request", async function (request) {
         msg = '{"type": "msg", "name": "' + name + '", "msg":"' + data.msg + '"}';
         break;
       case "raise":
-        if (!room.cards.find((item) => item.card === data.card && item.id === id)) {
+        if (!readOnly && !room.cards.find((item) => item.card === data.card && item.id === id)) {
```

Comment:
same as above

```suggestion
        if (readOnly) break;
        if (!room.cards.find((item) => item.card === data.card && item.id === id)) {

Reply:
I fail to see why this is the same case as above. The if-Statement is being close right before the break-Statement so it should habe the same effect as your suggested code.

Reply:
ah yes, `msg` is only set conditionally :+1: